sydent/replication/peer.py

```diff
@@ -258,7 +259,7 @@ def _pushSuccess(
             the status code.
         :type updateDeferred: twisted.internet.defer.Deferred
         """
-        if result.code >= 200 and result.code < 300:
+        if cast(int, result.code) >= 200 and cast(int, result.code) < 300:
```

Comment:
This is due to the `mypy-zope` plugin not being configured on this branch.

A basic one like:

```ini
[mypy]
plugins = mypy_zope:plugin
```

Has this line pass without needing the `cast` calls. Using `reveal_type` before including `mypy.ini` you get:

```
sydent/replication/peer.py:262: note: Revealed type is "zope.interface.interface.Attribute"
```

After including that you get:

```
sydent/replication/peer.py:262: note: Revealed type is "Any"
```

Ideally it would be good if that had a type on it, but I think that's due to `IResponse` not having a type, so I think we can safely ignore that for the moment.

Reply:
I'm a doofus; I had mypy-zope installed locally, but didn't add it to the mypy plugins. Fat lot of good that did!

Reply:
Commented as part of https://github.com/matrix-org/sydent/pull/357

Reply:
I reverted the change and will leave the addition of the plugin to the other PR you mentioned.